src/tests/End2EndTestsMain.cpp

```diff
@@ -16,7 +16,28 @@
 #include "tests/WebnnTest.h"
 
 int main(int argc, char** argv) {
-    InitWebnnEnd2EndTestEnvironment();
+    std::string device = "Default";
+    for (int i = 1; i < argc; ++i) {
+        if (strcmp("-d", argv[i]) == 0 && i + 1 < argc) {
+            device = argv[i + 1];
+        }
+    }
+    if (device != "CPU" && device != "GPU" && device != "Default") {
+        dawn::ErrorLog()
+            << "Invalid device options, only support \"CPU\", \"GPU\" and \"Default\".";
+        return -1;
```

Comment:
remove L.25 - L.29 and add the exception to L.39.

Reply:
Done, PTAL, thanks! @fujunwei